### DISCUSSION TITLE
[Feature Request] Utilize Embedded Sort Tags

### DISCUSSION THREAD

**Discussion by NeuroDawg** [navidrome/navidrome#2632] [Ideas]
I have taken my time to extensive curate my music, and ensure that the embedded tags are accurate, meaningful, and, hopefully, useful. I currently use Plex for my music server, but would really like to move away from that platform. I have installed Navidrome, but the one thing that keeps me from moving to it permanently is the fact that Navidrome reads, but doesn't utilize embedded sort tags.

I would like to see Navidrome give the option to use artistsort, albumartistsort, and albumsort tags for the organization of files. To either automatically sort using those tags, or give the option to use those tags rather than artist, albumartist, and album when music is imported into Navidrome.

Here is my reasoning. It's a kludge, but for my classical music I use the artist field to identify the performer (i.e. London Philharmonic, Kronos Quartet, etc.) and the album artist field for the composer. I like to be able to display by album artist and have composers ordered by last name, which is where the albumartistsort tag comes in handy. I really do prefer to see Alexander Borodin listed before Aaron Copland, as I always remember a composers last name. :)

For my rock/pop/jazz I also like to sort solo artists by last name. So I prefer to see Leonard Cohen before Bob Dylan. It's how I would organize my physical CDs, so when I go looking I know that Leonard Cohen is under the 'C's and no the 'L's.

Even the image from the record store used for Navidrome's subreddit header shows solo artists being sorted by last name. I'm of that generation, and am stuck in my ways.

**Comment by deluan** [comment]
Sorry for the looong delay. As you already [figured out](https://www.reddit.com/r/navidrome/comments/1eanvxo/sort_tags_now_utilized/), this was implemented a while ago, with the [PreferSortTags](https://www.navidrome.org/docs/usage/configuration-options/#:~:text=PreferSortTags-,ND_PREFERSORTTAGS,-Use%20Sort_*%20tags) option. Glad that you found it.

**Reply by ztx-lyghters** [reply]
Is the `albumsort` tag not implemented? I can't find a list of what tags are supported for sorting. I'm looking for a way to manually sort albums with identical release dates

**Reply by deluan** [reply]
Here's the complete list of default tags Navidrome imports on scan: https://github.com/navidrome/navidrome/blob/master/resources/mappings.yaml

For sort tags, look for the ones suffixed by `sort`. `albumsort` is definitely one of them.

**Reply by ztx-lyghters** [reply]
Hmm... But is it being utilized by Navidrome's web UI? I'm talking about `PreferSortTags` being set to true, for example. Navidrome doesn't change the album order on artist pages according to `albumsort` tag, so how is it being used exactly?

**Reply by deluan** [reply]
Yes, `albumsort` and all other sort tags should be used by the UI, if `PreferSortTags` is `true`. 

Related issue: https://github.com/navidrome/navidrome/issues/4907

**Reply by ztx-lyghters** [reply]
> Yes, `albumsort` and all other sort tags should be used by the UI, if `PreferSortTags` is `true`.
> 
> Related issue: #4907

Thanks, I might file a bug later then. Navidrome seems to ignore `albumsort` tags in my case. The order of albums with identical release dates stays the same on artist pages, no matter what I put in `albumsort`.
